src/iregnet_fit.cpp

```diff
@@ -538,3 +659,141 @@ compute_lambda_max(Rcpp::NumericMatrix X, double *w, double *z, double *eta,
 
   return lambda_max;
 }
+
+static int *
+sort_X_and_y(mat &sorted_X, mat &sorted_y, IREG_CENSORING *status, mat X, mat y, int function_type)
```

Comment:
missing docstring. Also, perhaps rename this to `sort_x_and_y_by_censoring_status` or something like that?